Create pod to force volume provisioning in dynamic provisioning storage e2e test

bertinatto opened the pull request:
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

The test creates a PVC and wait for it to be bound, however, it currently uses the cluster default StorageClass, which might be configured with `WaitForFirstConsumer` instead.

This issue was discovered in this job added recently: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-storage-selinux/1268090733013766146

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

CC @jsafrane @msau42

bertinatto commented:
/retest

oomichi commented:
/cc @oomichi
/retest

msau42 commented on test/e2e/storage/volume_provisioning.go:
```diff
@@ -662,10 +662,12 @@ var _ = utils.SIGDescribe("Dynamic Provisioning", func() {
 			}
-
+			suffix := fmt.Sprintf("volume-lifecycle")
+			test.Class = newStorageClass(test, ns, suffix)
```
What if we change the test to handle delayed binding instead?

bertinatto replied:
Added a commit to handle delayed binding.

From what I understand, this test should create its own SC anyway, so I left the first patch as well.

msau42 replied:
Then we're not actually testing the default StorageClass, which I think is the purpose of this test.

bertinatto replied:
I assumed the opposite, since other tests explicitly mention they use the default SC. Dropped this code.

bertinatto committed:
Create pod to force volume provisioning in storage e2e test

Otherwise, tests can fail if the default StorageClass
is configured with late binding.

bertinatto commented:
/retest

msau42 commented:
/retest
/lgtm
/approve

msau42 commented:
/hold
Can you update the PR title?

msau42 commented:
/priority important-soon

bertinatto commented:
/hold cancel
Updated title

bertinatto commented:
/retest

bertinatto commented:
/retest
